### PR TITLE
Add subgraph endpoint and query subdomains

### PR DESCRIPTION
**Pull request by nenadjaja**

### CODE REVIEW

**Comment by nenadjaja** [comment]
@jefflau When I try to run the app (from the `dev` branch), I am getting errors. When I rebased my code, I ran into the same errors, so I am not sure how to proceed. Am I missing some `ENV` setup (I sourced the `.env` file)

**Comment by nenadjaja** [comment]
Update - I was able to get it to work and deployed here: http://disillusioned-increase.surge.sh/ But I don't have the Infura project ID, so the Json RPC call fails..

**Comment by jefflau** [comment]
Hey @nenadjaja,

Thanks for doing this!

I'll review this code Monday next week as we have a holiday in Taiwan tomorrow. I'll try and run it locally, which should work fine with Infura

**Comment by nenadjaja** [comment]
Great, thank you @jefflau . Yeah, it works locally with Infura (on my machine). We might want to add `subdomains` in 1 other place in the UI (see screenshot bellow), but it could be redundant. Let's sync up on Monday, I am @nena on Discord. Happy Holiday! :)
Screen Shot 2019-06-06 at 9 30 04 AM

**Comment by jefflau** [comment]
Does the graph support ENS on any testnet? It would be nice to be able to test this on testnet, specifically adding subdomains.

I'd like to be to 'refetch' the query after I add a subdomain to a domain and see it reflected in the UI. Does this PR support this?

**Comment by nenadjaja** [comment]
Great questions @jefflau . See my answers below :)
> Does the graph support ENS on any testnet?
Yes, we do support kovan, rinkeby, ropsten and goerli testnets on our hosted service. Someone from your team would need to deploy the subgraph to one of those testnets, _I think_

> I'd like to be to 'refetch' the query after I add a subdomain to a domain and see it reflected in the UI. Does this PR support this?
This PR doesn't support it, good point. Adding subgraphs would be done using `web3` and then we’d refetch them basically using the subgraph’s API endpoint. I need to check where we are doing that, and I'll add the refetch logic

**Comment by jefflau** [comment]
@nenadjaja Thanks! Let me know if you have any questions

**Comment by nenadjaja** [comment]
Yeah do you have some time to sync up @jefflau ?

**Comment by jefflau** [comment]
Sure. You can get in touch with me at jeff [at] ens [dot] domains

Im available tomorrow day time GMT+8


Sent from my iPhone

On 11 Jun 2019, at 12:31 PM, Nena Djaja <notifications@github.com<mailto:notifications@github.com>> wrote:


Yeah do you have some time to sync up @jefflau<https://github.com/jefflau> ?

**Comment by nenadjaja** [comment]
Sent you an email from nena [at] thegraph [dot] com

**Comment by nenadjaja** [comment]
Also, I realized that we are already refetching the data from the subgraph after Adding a subdomain (that's in this PR, I just didn't notice it). I am not sure though how to test it.. can't see a way to add a subdomain

**Comment by nenadjaja** [comment]
@jefflau unfortunately I had to do a hack for testing - I am checking for the `networkId` and rendering the old code for testing, and the new one for local dev and prod :-/ At the moment, we don't have the solution for that problem, but we'll work on making it possible in the future. 
This runs on my local machine. Pls let me know if you need me to do anything else

**Comment by jefflau** [comment]
@nenadjaja Sorry I completely missed this comment. I think that's okay for now. I'll check the PR and hopefully get it merged in

**Comment by jefflau** [comment]
Hey @nenadjaja,

There is just one small thing that needs to be changed. We've talked about representing the names slightly differently depending on whether or not we know the labelhashes, previously I would just concenate the labelhash (0x123ecf...) to the name so you would end up with something like `0x123ecf.jefflau.eth`, however we now want undecrypted labelhashes to be represented with square brackets. I have a function that does it in my `ensdomains/ui` library which you can see here: https://github.com/ensdomains/ui/blob/master/src/registry.js#L351

Is there any way we can get the graph subnode to represent the names like this also?

**Comment by nenadjaja** [comment]
@jefflau I updated it, hope that's what you meant :) Btw, this https://github.com/ensdomains/ui/blob/master/src/registry.js#L351 doesn't take me to line `351` cause it's not there ;)

**Comment by jefflau** [comment]
Hey Nena,

That's not what I was talking about. It's related to how the data is coming back, so not related to the app. Basically I want the data to come back like this:

`[b2930b35844a230f00e51431acae96fe543a0347].vitalik.eth` assuming that the labelhash cannot be decrypted. The original labelhash would be 
`0xb2930b35844a230f00e51431acae96fe543a0347`.

**Comment by nenadjaja** [comment]
You mean how the data is coming back from the subgraph @jefflau ?

**Comment by jefflau** [comment]
@nenadjaja Yes exactly

**Comment by nenadjaja** [comment]
Ohh I think someone from your team created that subgraph and they would need to edit it. Someone from my team can help. Basically they would make the change, and deploy the subgraph again (as a new version), and then our endpoint will point to that new version once is indexed (synced up). Do you need us to help with the change?

**Comment by jefflau** [comment]
@nenadjaja That would be great yeah!

**Comment by Jannis** [comment]
Hi! I talked to @Arachnid and he offered make the necessary subgraph changes to support `[hash].foo.eth`. Let us know if we can help with anything else 😄

File: src/components/SingleName/SubDomains.js [hunk 1]
```diff
@@ -154,10 +154,7 @@ class SubDomains extends Component {
                             />
                             {d.labelName !== null
                               ? d.name
-                              : `[unknown${d.labelhash.slice(
```
Comment (jefflau) [inline src/components/SingleName/SubDomains.js]: Can we change this back?

**Comment by jefflau** [comment]
@Arachnid said he deployed the ropsten subgraph as `ens/ensRopsten`

**Pull request closed without merging**